module/thermal_mgmt/src/mod_thermal_mgmt.c

```diff
@@ -34,6 +34,68 @@ struct mod_thermal_mgmt_actor_ctx *get_actor_ctx(
     return &dev_ctx->actor_ctx_table[actor];
 }
 
+
+#ifdef PID
```

Comment:
IMO there's no need to create two implementations that are almost identical but the kd part.
I would suggest to add the derivative part in the pi_control as is, no additional #IFDEF.
Platforms that do not care about the derivative part can simply leave that coefficient to zero.

Then rename pi_controller() to pid_controller() please